app/controllers/pages_controller.rb

```diff
@@ -1,6 +1,8 @@
 class PagesController < ApplicationController
   def landing
     params[:id] = 'landing'
+    @signed_in = params[:signed_in]
+    @previous_course = courses_service.get params[:course_id] if params.key?(:course_id)
```

Comment:
Need some error handling here in case `course_id` is not found (probably unlikely, but would be good to fail softly and still render the landing page regardless).